Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-20-0-48.ca-central-1.compute.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.18.0,  cUrl: 7.67.0,  node.js: v16.13.0,  npm: 8.1.0
Sha: e2f4a173f11702ee48f43a96fb1f6e58a49b9520
Containerized:False
Number of Error(s): 4
/home/centos/smoketest/PR-15536/HPCC-Platform/system/metrics/sinks/prometheus/prometheusSink.cpp:109:13: error: expected \xe2\x80\x98;\xe2\x80\x99 before \xe2\x80\x98is\xe2\x80\x99
make[2]: *** [system/metrics/sinks/prometheus/CMakeFiles/hpccmetrics_prometheussink.dir/build.make:82: system/metrics/sinks/prometheus/CMakeFiles/hpccmetrics_prometheussink.dir/prometheusSink.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:17705: system/metrics/sinks/prometheus/CMakeFiles/hpccmetrics_prometheussink.dir/all] Error 2
make: *** [Makefile:182: all] Error 2
Get ECLWatch build result
ReportTimes.
Finish at: 2021-11-02_04-11-04
time logs started
Prep time   :    16 sec (00:00:16)
Build time  :   137 sec (00:02:17)
Package time:     0 sec (00:00:00)
Install time:     0 sec (00:00:00)
Start time  :     0 sec (00:00:00)
Test time   :     0 sec (00:00:00)
Stop time   :     0 sec (00:00:00)
Summary     :   153 sec (00:02:33)
time logs end

[Build error:](https://gist.github.com/2b0b7615272e5f665980147632345ac6)
Build log [Build log](https://gist.github.com/HPCCSmoketest/2b0b7615272e5f665980147632345ac6#file-result-21-11-02-04-08-18-log)

--- a/system/metrics/sinks/prometheus/prometheusSink.cpp
+++ b/system/metrics/sinks/prometheus/prometheusSink.cpp
@@ -104,7 +104,10 @@ void PrometheusMetricSink::toPrometheusMetrics(const std::vector<std::shared_ptr
 
     for (auto &pMetric: reportMetrics)
     {
-        const std::string & name = pMetric->queryName();
+        std::string name = pMetric->queryName();
+
+        //'.' is a known char used in HPCC metric names but invalid in Prometheus
+        std::replace(name.begin(), name.end(), '.', '_');
         if (verbose)
         {
             if (!pMetric->queryDescription().empty())

